app.ts

```diff
@@ -0,0 +1,323 @@
+		})
+		.then(() => {
+			$("#loader").hide();
+			$("#tableWrapper").show();
+			$("#prevPageButton").show();
+			if (searchPage === totalPages) {
+				$("#nextPageButton").hide();
+			} else {
+				$("#nextPageButton").show();
+			}
+		}).catch((error) => {
+			throw error;
+		})
+}
+
+// Update the screen layout and display records based on the current screen/window size
+async function updateScreen() {
+	const newScreenHeight = window.innerHeight;
+	totalRecords();
+	state.recordsPerPage = screenCalculations(newScreenHeight);
+	$("#loader").show();
+	$("#tableWrapper").hide();
+	if (state.searchedIndex !== null) {
+		const searchPage = Math.ceil((state.searchedIndex + 1) / state.recordsPerPage);
+		const searchPageIndex = state.searchedIndex % state.recordsPerPage;
+		state.searchedIndex = (searchPage - 1) * state.recordsPerPage + searchPageIndex;
+		displayData(Math.max(state.searchedIndex - state.recordsPerPage + 1, 0), state.recordsPerPage);
+	} else {
+		const previousFirstRecordIndex = state.currentFirstRecordIndex;
+		currentPage = Math.ceil((state.currentFirstRecordIndex + 1) / state.recordsPerPage);
+		displayData(state.currentFirstRecordIndex, state.recordsPerPage);
+	}
+	$("#loader").hide();
+	$("#tableWrapper").show();
+	if (currentPage === totalPages) {
+		$("#nextPageButton").hide();
+	} else {
+		$("#nextPageButton").show();
+	}
+	if (state.currentFirstRecordIndex <= 1) {
+		$("#prevPageButton").hide();
+	} else {
+		$("#prevPageButton").show();
+	}
+}
+
+window.onload = async function () {
+	$("#loader").hide();
+	updateScreen();
+	fetchColumns();
+	inputHandling();
+	$(window).on("resize", debounce(async () => {
+		$("#tableWrapper").hide();
+		$("#loader").show();
+		updateScreen()
+			.then(async () => {
+				const totalRecCount = await totalRecords();
+				if (currentPage * state.recordsPerPage > totalRecCount - 1) {
+					const lastPage = Math.ceil(totalRecCount / state.recordsPerPage);
+					currentPage = lastPage;
+					state.currentFirstRecordIndex = (lastPage - 1) * state.recordsPerPage;
+					return displayData(state.currentFirstRecordIndex, state.recordsPerPage);
+				}
+			})
+			.then(() => {
+				$("#loader").hide();
+				$("#tableWrapper").show();
+			}).catch((error) => {
+				throw (error);
+			});
+	}, 150))
+
+	$("#searchInput").on("keydown", function (e) {
+		if (e.key === "e" || e.key === "E" || e.key === "."
+			|| e.key === "+" || e.key === "-") {
+			e.preventDefault();
+		}
+	})
+
+	$("#searchForm").submit(async function (e) {
+		e.preventDefault();
+		const searchValue = $("#searchInput").val();
+		$("#tableWrapper").hide();
+		$("#loader").show();
+		state.searchedIndex = null;
+		await searchMethod(searchValue);
+		if (state.searchedIndex !== null) {
+			currentPage = Math.ceil((state.searchedIndex + 1) / state.recordsPerPage);
+			state.currentFirstRecordIndex = Math.max(state.searchedIndex - state.recordsPerPage + 1, 0);
+		};
+		$("#loader").hide();
+		$("#tableWrapper").show();
```

Comment:
```suggestion
		searchMethod(searchValue)
		.then(() => {
				if (state.searchedIndex !== null) {
			currentPage = Math.ceil((state.searchedIndex + 1) / state.recordsPerPage);
			state.currentFirstRecordIndex = Math.max(state.searchedIndex - state.recordsPerPage + 1, 0);
		}
		$("#loader").hide();
		$("#tableWrapper").show();
		});
```
Handle your promise rather. It feels cleaner than `await`. Excuse the indentation in the suggestion.